internal/internal_task_handlers.go

```diff
@@ -274,51 +266,29 @@ OrderEvents:
 		switch event.GetEventType() {
 		case s.EventTypeDecisionTaskStarted:
 			if !eh.IsNextDecisionFailed() {
-				eh.currentIndex++ // Since we already processed the current event
-				decisionStartedEvent = event
+				eh.currentIndex++
+				reorderedEvents = append(reorderedEvents, event)
```

Comment:
May be rename "reorderedEvents" to something else as there is no more reordering.